cmd/revendor/run-glide.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Masterminds/glide/action"
+	"github.com/Masterminds/glide/repo"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+func main() {
+	var (
+		pkg      string
+		revision string
+	)
+	flag.StringVar(&pkg, "package", "", "name of the vendored package to update")
+	flag.StringVar(&revision, "revision", "", "revision to which to update")
+	flag.Parse()
+	if pkg == "" || revision == "" {
+		grip.Info("no repo to revendor, run-glide will no-op")
+		return
+	}
+
+	updateGlide(pkg, revision)
+
+	installer := repo.NewInstaller()
+	action.EnsureGoVendor()
+	action.Install(installer, false, false)
+}
+
+func updateGlide(pkg, revision string) {
+	wd, err := os.Getwd()
+	if err != nil {
+		grip.Error(errors.Wrap(err, "error getting working dir"))
+		return
+	}
+	vendorPath := filepath.Join(wd, "vendor", pkg)
+	_, err = os.Stat(vendorPath)
+	grip.EmergencyFatal(errors.Wrapf(err, "vendor directory %s not found", vendorPath))
+	glidePath := filepath.Join(wd, "glide.lock")
+	_, err = os.Stat(glidePath)
+	grip.EmergencyFatal(errors.Wrapf(err, "glide file %s not found", glidePath))
+
+	glideFile, err := os.Open(glidePath)
+	grip.EmergencyFatal(errors.Wrap(err, "error opening glide file"))
+	glide, err := ioutil.ReadAll(glideFile)
+	grip.EmergencyFatal(errors.Wrap(err, "error reading glide file"))
+	lines := strings.Split(string(glide), "\n")
+	found := false
+	for i, line := range lines {
+		if strings.Contains(line, pkg) {
+			lines[i+1] = fmt.Sprintf("    version: %s", revision)
+			found = true
+			break
+		}
+	}
+	if !found {
+		grip.EmergencyFatal("error reading glide file")
+	}
+	err = ioutil.WriteFile(glidePath, []byte(strings.Join(lines, "\n")), 0777)
+	grip.EmergencyFatal(errors.Wrap(err, "error writing glide file"))
+
+	err = os.RemoveAll(vendorPath)
+	grip.EmergencyFatal(errors.Wrap(err, "error removing vendor dir"))
```

Comment:
I'd combine 66 and 67 into one line and the same with 69 and 70

Reply:
done